secure_message_v2/controllers/messages.py

```diff
@@ -0,0 +1,38 @@
+import logging
+from datetime import datetime
+
+import structlog
+
+from secure_message_v2.controllers.threads import update_read_status
+from secure_message_v2.models.models import Message
+from secure_message_v2.utils.session_decorator import with_db_session
+
+logger = structlog.wrap_logger(logging.getLogger(__name__))
+
+
+@with_db_session
+def post_new_message(posted_message, session):
+    """
+    Post a message in an existing thread
+    :param message: the message to be created
+    :param session
+    :return: the created message object
+    """
+    current_time = datetime.utcnow()
+
+    message = Message(
+        thread_id=posted_message["thread_id"],
+        body=posted_message["body"],
+        is_from_internal=posted_message["is_from_internal"],
+        sent_by=posted_message["sent_by"],
+        sent_at=current_time,
+    )
+
+    session.add(message)
+    session.flush()
+
+    update_read_status(
+        message.thread_id, message.is_from_internal, not message.is_from_internal
```

Comment:
This is futureproofing - at some point we'll definitely have an endpoint to update one or the other (and not necessarily to opposite values) - otherwise I'd agree